core/src/main/java/org/kiwix/kiwixmobile/core/page/bookmark/BookmarksFragment.kt

```diff
@@ -19,8 +20,8 @@ class BookmarksActivity : PageActivity() {
   override val noItemsString: String by lazy { getString(R.string.no_bookmarks) }
   override val switchString: String by lazy { getString(R.string.bookmarks_from_current_book) }
   override val switchIsChecked: Boolean by lazy { sharedPreferenceUtil.showBookmarksAllBooks }
-  override fun injection(coreComponent: CoreComponent) {
-    activityComponent.inject(this)
+  override fun inject(baseActivity: BaseActivity) {
+    (requireActivity() as CoreMainActivity).cachedComponent.inject(this)
```

Comment:
probably an extension func to get `cachedComponent` to keep the casting isolated in 1 place

Reply:
Nice idea!